packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -14,195 +27,109 @@ import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/compone
+  const hideControls = !isPlaying && !showControlsWhenPaused
 
   return (
-    <div
-      className={clsx(
-        styles.VideoPlayer__container,
-        styles.VideoPlayer__overlays,
-        !playing && styles.VideoPlayer__showOverlays,
-      )}
-      ref={videoWrapperRef}
-    >
-      <video
-        ref={videoRef}
-        title={title}
-        controls={false}
-        poster={poster}
-        className={clsx(styles.VideoPlayer, className)}
-        onPlay={handleOnPlay}
-        onPause={handleOnPause}
-        onPlaying={handleOnPlaying}
-        onLoadedMetadata={handleOnLoadedMetadata}
-        onTimeUpdate={handleOnTimeUpdate}
-        {...rest}
-      >
+    <div className={styles.VideoPlayer__container} ref={videoWrapperRef}>
+      <video ref={ref} title={title} controls={false} className={clsx(styles.VideoPlayer, className)} {...rest}>
         {children}
         <track kind="captions" />
       </video>
-      {title && (
-        <div className={styles.VideoPlayer__title}>
-          {branding && <MarkGithubIcon size={48} />}
-          <Text size="500" weight="medium" className={styles.VideoPlayer__controlTextColor}>
+      <div className={styles.VideoPlayer__title}>
+        {showBranding && <MarkGithubIcon size={40} />}
+        {!visuallyHiddenTitle && (
+          <Text size="400" weight="medium" className={styles.VideoPlayer__controlTextColor}>
             {title}
           </Text>
-        </div>
-      )}
-      <button className={styles.VideoPlayer__playButton} onClick={handleVideoPlayback}>
-        {!playing && (
-          <span className={styles.VideoPlayer__playButtonInner}>
-            <svg
-              xmlns="http://www.w3.org/2000/svg"
-              width="24"
-              height="24"
-              viewBox="0 0 24 24"
-              fill="none"
-              aria-hidden="true"
-            >
-              <path
-                d="M1.56489 23.8112C0.884817 24.2389 1.07491e-06 23.7501 1.0398e-06 22.9467L8.28679e-08 1.05473C4.73246e-08 0.241596 0.904067 -0.245404 1.58307 0.201969L18.5829 11.4026C19.2032 11.8113 19.1935 12.7244 18.5647 13.1198L1.56489 23.8112Z"
-                fill="currentColor"
-              />
-            </svg>
-          </span>
         )}
-        <span className="visually-hidden">Play</span>
+      </div>
+      <button
+        className={styles.VideoPlayer__playButton}
+        onClick={togglePlaying}
+        aria-label={isPlaying ? 'Pause' : 'Play'}
+      >
+        {isPlaying ? null : <VideoPlayer.PlayIcon className={styles.VideoPlayer__playButtonOverlay} />}
       </button>
-      <Controls
-        videoRef={videoRef}
-        videoWrapperRef={videoWrapperRef}
-        totalTime={totalTime}
-        currentTime={playedTime}
-        playing={playing}
-        trackInformation={trackInformation}
-      />
+      <div className={styles.VideoPlayer__controls}>
+        {ccEnabled ? <Captions /> : null}
+        {hideControls ? null : (
```

Comment:
Nit but would this be more readable?

```suggestion
        {!hideControls && (
```

Reply:
37031f7e2b921e6708552ce76d27952885efdd4c